exonum/src/api/with.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2018 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use futures::Future;
+
+use std::marker::PhantomData;
+
+use super::{error, ServiceApiState};
+
+/// Type alias for the usual synchronous result.
+pub type Result<I> = ::std::result::Result<I, error::Error>;
+/// Type alias for the asynchronous result that will appear in future.
```

Comment:
I guess `appear` isn't the right word here... Perhaps `will be ready`? And `in the future`